src/main/drivers/adc_stm32f10x.c

```diff
@@ -29,17 +29,44 @@
 
 #include "sensor.h"
 #include "accgyro.h"
-
 #include "adc.h"
 #include "adc_impl.h"
+#include "io.h"
+#include "rcc.h"
 
 #ifndef ADC_INSTANCE
-#define ADC_INSTANCE                ADC1
-#define ADC_ABP2_PERIPHERAL         RCC_APB2Periph_ADC1
-#define ADC_AHB_PERIPHERAL          RCC_AHBPeriph_DMA1
-#define ADC_DMA_CHANNEL             DMA1_Channel1
+#define ADC_INSTANCE   ADC1
 #endif
 
+const adcDevice_t adcHardware[] = { 
+    { .ADCx = ADC1, .rccADC = RCC_APB2(ADC1), .rccDMA = RCC_AHB(DMA1), .DMAy_Channelx = DMA1_Channel1 }  
+};
+
+ADCDevice adcDeviceByInstance(ADC_TypeDef *instance)
+{
+    if (instance == ADC1) 
+        return ADCDEV_1;
+
+/* TODO -- ADC2 available on large 10x devices.
+    if (instance == ADC2)
+        return ADCDEV_2;
+*/
+    return ADCINVALID;
+}
+
+const adcPinMap_t adcPinMap[] = { 
+    { .pin = DEFIO_TAG_E__PA0, .channel = ADC_Channel_0 }, // ADC12
+    { .pin = DEFIO_TAG_E__PA1, .channel = ADC_Channel_1 }, // ADC12
+    { .pin = DEFIO_TAG_E__PA2, .channel = ADC_Channel_2 }, // ADC12
+    { .pin = DEFIO_TAG_E__PA3, .channel = ADC_Channel_3 }, // ADC12
+    { .pin = DEFIO_TAG_E__PA4, .channel = ADC_Channel_4 }, // ADC12
+    { .pin = DEFIO_TAG_E__PA5, .channel = ADC_Channel_5 }, // ADC12
+    { .pin = DEFIO_TAG_E__PA6, .channel = ADC_Channel_6 }, // ADC12
+    { .pin = DEFIO_TAG_E__PA7, .channel = ADC_Channel_7 }, // ADC12
+    { .pin = DEFIO_TAG_E__PB0, .channel = ADC_Channel_8 }, // ADC12
+    { .pin = DEFIO_TAG_E__PB1, .channel = ADC_Channel_9 }, // ADC12   
+};
```

Comment:
I wouldn't bother with the item names here, ie
```
const adcPinMap_t adcPinMap[] = {
    { DEFIO_TAG_E__PA0, ADC_Channel_0 }, // ADC12
    { DEFIO_TAG_E__PA1, ADC_Channel_1 }, // ADC12
...
``` 
conveys just as much information.